### PR TITLE
Add IPFS as a new topic

### PR DESCRIPTION
**Pull request by VictorBjelkholm**
- [X] I followed the contributing guidelines: https://github.com/github/explore/blob/master/CONTRIBUTING.md

I am:
  - [ ] Suggesting edits to an existing Topic page
  - [X] Curating a new Topic page

************CURATING A NEW TOPIC PAGE************

- [X] I've formatted my changes as a new folder directory, named for the topic as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]`)
- [X] My folder contains a `*.png` image (if applicable) and `index.md`
- [X] All required fields in my `index.md` conform to the Style Guide and API docs: https://github.com/github/explore/tree/master/docs

Please explain why you think this Topic page should be curated:

Lots of projects are using the IPFS topic but it doesn't have any more information about what it is. At the time of opening this PR, 248 projects are using the IPFS topic in their repository.

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

**Comment by VictorBjelkholm** [comment]
Seeing CI failed, will fix the cases, but one of them don't make much sense.

```
  2) Failure:
topics::ipfs topic#test_0011_has valid aliases [/home/travis/build/github/explore/test/topics_test.rb:125]:
'InterPlanetary-FileSystem' must be between 1-35 characters, start with a letter or number, and may include hyphens
```

The string 'InterPlanetary-FileSystem' is 25 characters long...

File: topics/ipfs/index.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+aliases: InterPlanetary-FileSystem, InterPlanetary FileSystem, Inter-Planetary FileSystem, IPFS, ipfs, InterPlanetary File System
+created_by: Juan Benet 
+display_name: IPFS
+logo: ipfs.png
+released: May 23, 2014
+short_description: InterPlanetary File System (IPFS) is a protocol and network designed to create a content-addressable, peer-to-peer method of storing and sharing hypermedia in a distributed file system
+topic: IPFS
+url: https://ipfs.io/
```
Comment (cheshire137) [inline topics/ipfs/index.md]: Can you add a line with `github_url: https://github.com/ipfs/ipfs`?

File: topics/ipfs/index.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+---
+aliases: InterPlanetary-FileSystem, InterPlanetary FileSystem, Inter-Planetary FileSystem, IPFS, ipfs, InterPlanetary File System
```
Comment (cheshire137) [inline topics/ipfs/index.md]: I checked on these and they don't look to be in use, or are the same as the topic `ipfs` itself. Can you remove this line? https://github.com/topics/interplanetary-file-system, https://github.com/topics/inter-planetary-filesystem, and https://github.com/topics/interplanetary-filesystem have no repositories.

If you want, you could add different aliases for topics that revolves around IPFS, where having info about IPFS would be useful to provide context, e.g.,

```yaml
aliases: ipfs-daemon, ipfs-api, ipfs-webui, js-ipfs, ipfs-protocol, go-ipfs
```

File: topics/ipfs/index.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+wikipedia_url: https://en.wikipedia.org/wiki/InterPlanetary_File_System
+---
+
+IPFS is a peer-to-peer distributed file system that seeks to connect all computing devices with the same system of files.
+In some ways, IPFS is similar to the World Wide Web, but IPFS could be seen as a single BitTorrent swarm, exchanging
+objects within one Git repository. In other words, IPFS provides a high-throughput, content-addressed block storage model,
+with content-addressed hyperlinks. This forms a generalized Merkle directed acyclic graph (DAG). IPFS combines a distributed
+hash table, an incentivized block exchange, and a self-certifying namespace. IPFS has no single point of failure, and nodes
+do not need to trust each other, except for every node they are connected to. Distributed Content Delivery saves bandwidth and
+prevents DDoS attacks, which HTTP struggles with.
```
Comment (cheshire137) [inline topics/ipfs/index.md]: > Distributed Content Delivery saves bandwidth and prevents DDoS attacks, which HTTP struggles with.

Maybe rephrase to:

> Distributed Content Delivery saves bandwidth and prevents distributed denial-of-service (DDoS) attacks, a problem with HTTP.

File: topics/ipfs/index.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+display_name: IPFS
+logo: ipfs.png
+released: May 23, 2014
+short_description: InterPlanetary File System (IPFS) is a protocol and network designed to create a content-addressable, peer-to-peer method of storing and sharing hypermedia in a distributed file system
+topic: IPFS
+url: https://ipfs.io/
+wikipedia_url: https://en.wikipedia.org/wiki/InterPlanetary_File_System
+---
+
+IPFS is a peer-to-peer distributed file system that seeks to connect all computing devices with the same system of files.
```
Comment (cheshire137) [inline topics/ipfs/index.md]: Can you join lines 13-19? This body will be rendered as Markdown so every line break you have will come out as a `<br>` HTML tag.

File: topics/ipfs/index.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+---
+aliases: InterPlanetary-FileSystem, InterPlanetary FileSystem, Inter-Planetary FileSystem, IPFS, ipfs, InterPlanetary File System
+created_by: Juan Benet 
+display_name: IPFS
+logo: ipfs.png
+released: May 23, 2014
+short_description: InterPlanetary File System (IPFS) is a protocol and network designed to create a content-addressable, peer-to-peer method of storing and sharing hypermedia in a distributed file system
```
Comment (cheshire137) [inline topics/ipfs/index.md]: Can you add a period to the end?

**Comment by cheshire137** [comment]
> 'InterPlanetary-FileSystem' must be between 1-35 characters, start with a letter or number, and may include hyphens

We should update the error message. It's complaining because the alias needs to be lowercase.

File: topics/ipfs/index.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+topic: IPFS
```
Comment (cheshire137) [inline topics/ipfs/index.md]: Can you make this lowercase? `topic: ipfs`

Comment (VictorBjelkholm) [reply topics/ipfs/index.md]: Fixed

File: topics/ipfs/index.md (hunk 3 above)
Comment (VictorBjelkholm) [reply topics/ipfs/index.md]: Fixed

File: topics/ipfs/index.md (hunk 4 above)
Comment (VictorBjelkholm) [reply topics/ipfs/index.md]: Have not used a markdown before that behaves that way. Fixed this now.

Reference: 
image

https://help.github.com/articles/basic-writing-and-formatting-syntax/#paragraphs-and-line-breaks

File: topics/ipfs/index.md (hunk 1 above)
Comment (VictorBjelkholm) [reply topics/ipfs/index.md]: Done

File: topics/ipfs/index.md (hunk 5 above)
Comment (VictorBjelkholm) [reply topics/ipfs/index.md]: Done

File: topics/ipfs/index.md (hunk 2 above)
Comment (VictorBjelkholm) [reply topics/ipfs/index.md]: Yeah, fixed this. My thinking was to avoid future topics to not be linked, but what you're saying is correct, those aliases are not used.

**Comment by VictorBjelkholm** [comment]
@cheshire137 thanks a lot for all the pointers. I've now addressed all the comments and CI is passing.

File: topics/ipfs/index.md (hunk 4 above)
Comment (cheshire137) [reply topics/ipfs/index.md]: Might be a quirk of how we're rendering it. Check out the screenshot in https://github.com/github/explore/pull/129 for how the body was rendering when it had line breaks, to see an example.

Comment (VictorBjelkholm) [reply topics/ipfs/index.md]: Strange huh, are you not using GFM on Topics like on the rest of Github?

**Comment by VictorBjelkholm** [comment]
Seeing the page now, I realize the description is waaaay too long.

**Comment by VictorBjelkholm** [comment]
@cheshire137 would it be fine to just have the description and short_description be the same?

**Comment by cheshire137** [comment]
Hmm, the description and short_description should differ. I think the description IPFS has is a fine length, though it could be split into a couple paragraphs. Check out https://github.com/topics/userscript, https://github.com/topics/xamarin, or https://github.com/topics/babel for example.

**Comment by VictorBjelkholm** [comment]
@cheshire137 ok, that makes sense. Thanks for guiding me, will have a think about it and hopefully send a patch with some updated formatting.
